packages/riipen-ui/src/components/Tooltip.jsx

```diff
@@ -8,130 +8,35 @@ import withClasses from "../utils/withClasses";
-     */
-    children: PropTypes.node,
-
-    /**
-     * Any additional classes to apply.
-     */
-    classes: PropTypes.array,
-
-    /**
-     * Whether tooltip should display on click.
-     */
-    click: PropTypes.bool,
-
-    /**
-     * Color of tooltip.
-     */
-    color: PropTypes.oneOf([
-      "default",
-      "white",
-      "positive",
-      "negative",
-      "warning"
-    ]),
-
-    /**
-     * The component used for the root node.
-     * Either a string to use a DOM element or a component.
-     */
-    component: PropTypes.elementType,
-
-    /**
-     * Function to call on tooltip keydown.
-     */
-    onKeyDown: PropTypes.func,
-
-    /**
-     * Whether tooltip should display on hover.
-     */
-    hover: PropTypes.bool,
-
-    /**
-     * How the open/close state will be determined:
-     * [Default] false: by this component's state,
-     * true: by this component's props.
-     */
-    isControlledByProps: PropTypes.bool,
-
-    /**
-     * Whether the popover should stay open after leaving the content.
-     */
-    keepOpenOnMouseLeave: PropTypes.bool,
-
-    /**
-     * Function to call on tooltip close.
-     */
-    onClose: PropTypes.func,
-
-    /**
-     * Function to call on tooltip open.
-     */
-    onOpen: PropTypes.func,
-
-    /**
-     * Whether the popover should be open/shown.
-     * Only used when props.isControlledByProps is true
-     * [Default] false
-     */
-    open: PropTypes.bool,
-
-    /**
-     * Where to display the tooltip in relation to element.
-     */
-    position: PropTypes.oneOf([
-      "top-right",
-      "top-center",
-      "top-left",
-      "center-right",
-      "center-left",
-      "bottom-right",
-      "bottom-center",
-      "bottom-left"
-    ]),
-
-    /**
-     * The size of the tooltip.
-     */
-    size: PropTypes.oneOf(["small", "medium"]),
-
-    /**
-     * The tooltip content to display.
-     */
-    tooltip: PropTypes.node
-  };
-
-  static defaultProps = {
-    classes: [],
-    click: false,
-    color: "default",
-    component: "div",
-    hover: true,
-    isControlledByProps: false,
-    open: false,
-    keepOpenOnMouseLeave: false,
-    position: "bottom-center",
-    size: "small"
-  };
-
-  constructor() {
-    super();
-    this.tooltipRootRef = React.createRef();
-    this.state = {
-      open: false
-    };
-  }
-
-  getLinkedStyles = () => {
-    const theme = this.context;
+const Tooltip = ({
+  children,
+  classes,
+  click,
+  color,
+  component: Component,
+  hover,
+  isControlledByProps,
+  keepOpenOnMouseLeave,
+  onClose,
+  onKeyDown,
+  onOpen,
+  open,
+  position,
+  size,
+  tooltip,
+  ...other
+}) => {
+  const theme = React.useContext(ThemeContext);
+  const [isOpen, setIsOpen] = useState(isControlledByProps ? open : false);
+  const tooltipRootRef = useRef(null);
+
+  useEffect(() => {
+    if (isControlledByProps) {
+      setIsOpen(open);
+    }
+  }, [isControlledByProps, open]);
```

Comment:
Just update state to open prop as needed. Cleans up the logic down the line a bit.